tools/lox_gen/src/main.rs

```diff
@@ -305,3 +305,72 @@ fn point_mass(
         }
     };
 }
+
+/// Generates implementations for [lox_core::bodies::RotationalElements].
+fn rotational_elements(
+    imports: &mut Vec<Ident>,
+    code: &mut TokenStream,
+    tests: &mut TokenStream,
+    ident: &Ident,
+    id: &i32,
+    data: &Data,
+) {
+    let shared_imports = vec![
+        format_ident!("RotationalElements"),
+        format_ident!("RACoefficients")
+    ];
+
+    for import in shared_imports {
+        if !imports.contains(&import) {
+            imports.push(import)
+        }
+    }
+
+    let (ra_p0, ra_p1) = if let Some(rac) = get_ra_coefficients(id, data) {
+        rac
+    } else {
+        return; // RotationalElements can't be implemented for this body
+    };
+
+    *code = quote! {
+        #code
+
+        impl RotationalElements for #ident {
+            const RIGHT_ASCENSION_COEFFICIENTS: RACoefficients = [#ra_p0, #ra_p1];
+        }
+    };
+
+    let ra_test_name = format_ident!("test_ra_coefficients_{}", *id as u32);
+    *tests = quote! {
+        #tests
+
+        #[test]
+        fn #ra_test_name() {
+            assert_eq!([#ra_p0, #ra_p1], #ident::RIGHT_ASCENSION_COEFFICIENTS)
+        }
+    }
+}
+
+fn get_ra_coefficients(id: &i32, data: &Data) -> Option<(f64, f64)> {
+    let key = format!("BODY{id}_POLE_RA");
+    match data.pck.get_double_array(&key) {
+        None => None,
+        Some(polynomials) if polynomials.len() < 2 => {
+            panic!(
```

Comment:
I recommend refactoring lox_gen's error handling so we have no panics and only one point of exit (from main). As the generator grows and we discover more error cases, panicking will become hard to maintain. Happy to do that in another PR if you agree.